pkgpanda/docker/dcos-builder/Dockerfile

```diff
@@ -49,4 +50,18 @@ RUN ln -sf /usr/bin/cpp-4.8 /usr/bin/cpp && \
   ln -sf /usr/bin/gcc-ranlib-4.8 /usr/bin/gcc-ranlib && \
   ln -sf /usr/bin/gcov-4.8 /usr/bin/gcov
 
+ENV GOLANG_VERSION 1.7.3
+ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
+ENV GOLANG_DOWNLOAD_SHA256 508028aac0654e993564b6e2014bf2d4a9751e3b286661b0b0040046cf18028e
+
+RUN curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
+  && echo "$GOLANG_DOWNLOAD_SHA256  golang.tar.gz" | sha256sum -c - \
+  && tar -C /usr/local -xzf golang.tar.gz \
+  && rm golang.tar.gz
+
+ENV GOPATH /go
+ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
```

Comment:
This is from https://github.com/docker-library/golang/blob/4fd5df86eea53623b1009b3621b40a97f9f359e5/1.7/Dockerfile#L22

If I had to guess:

/usr/local/go/bin goes in the PATH for the Go tools itself

The GOPATH stuff is there for downstream projects that want to build packages inside of the Go Docker. Hence all the go-wrapper stuff that was doing magic with /go/src/app:

https://github.com/docker-library/golang/blob/4fd5df86eea53623b1009b3621b40a97f9f359e5/1.7/go-wrapper

Seems like all our packages currently do some variant of export GOPATH=$GOPATH:/pkg. I think they should just do export GOPATH=/pkg and we can probably just not set GOPATH here.